### PR TITLE
Speed up the loading of the all watcher.

### PR DESCRIPTION
**Pull request by howbazaar**
When the all watcher starts up, it loads all of the entities and sets up the multiwatcher store. There are a bunch of subordinate collections that are read for every individual primary entity. This leads to many small reads. The intent of this branch is introduce a context object that may precache the subordinate values to make it faster to load them.

Unfortunately there are a bunch of things that we can't yet optimize. Primarily the open ports and network address bits. In order to avoid additional database access we need to introduce ordering of collection processing, and even ordering within the collection, so we can process principals before we process subordinates. This way instead of reading the db to find the principal, we could get it out of the multiwatcher store to get the assigned machine.

These additional changes will require another pass through the code to optimize.

## QA steps

```sh
deploy any reasonable sized bundle, as bundle deployments use a multiwatcher
```

## Documentation changes

None, internal change only.

### CODE REVIEW

**Comment by babbageclunk** [comment]
Looks like you have a couple of test failures though.

File: state/allwatcher.go [hunk 1]
```diff
@@ -949,11 +882,11 @@ func (s *backingStatus) updated(st *State, store *multiwatcherStore, id string)
 		newInfo := *info
 		// Get the unit's current recorded status from state.
 		// It's needed to reset the unit status when a unit comes off error.
-		statusInfo, err := getStatus(st.db(), unitGlobalKey(newInfo.Name), "unit")
+		statusInfo, err := ctx.getStatus(unitGlobalKey(newInfo.Name), "unit")
 		if err != nil {
 			return err
 		}
-		if err := s.updatedUnitStatus(st, store, id, statusInfo, &newInfo); err != nil {
+		if err := s.updatedUnitStatus(ctx, statusInfo, &newInfo); err != nil {
```
[the remark this answers is not in the document]
Comment (howbazaar) [reply state/allwatcher.go]: This is handled in the updatedUnitStatus method.

**Comment by howbazaar** [comment]
$$merge$$

**Comment by howbazaar** [comment]
$$merge$$

**Pull request merged**